src/main/java/org/broadinstitute/dsm/db/dao/kit/BSPKitDao.java

```diff
@@ -0,0 +1,231 @@
+                            String message = "Kit of participant " + bspKitQueryResult.getBspParticipantId() + " was received by GP. <br> " +
+                                    "CollaboratorSampleId:  " + bspKitQueryResult.getBspSampleId() + " <br> " +
+                                    received.getValue();
+                            notificationUtil.sentNotification(bspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.UNIVERSAL_NOTIFICATION_TEMPLATE, NotificationUtil.DSM_SUBJECT);
+                        }
+                        else {
+                            logger.error("Instance settings behavior for kit was not known " + received.getType());
+                        }
+                    }
+                    else {
+                        triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+                    }
+                }
+            }
+            else {
+                triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+            }
+
+            String bspParticipantId = bspKitQueryResult.getBspParticipantId();
+            String bspSampleId = bspKitQueryResult.getBspSampleId();
+            String bspMaterialType = bspKitQueryResult.getBspMaterialType();
+            String bspReceptacleType = bspKitQueryResult.getBspReceptacleType();
+            int bspOrganism;
+            try {
+                bspOrganism = Integer.parseInt(bspKitQueryResult.getBspOrganism());
+            }
+            catch (NumberFormatException e) {
+                throw new RuntimeException("Organism " + bspKitQueryResult.getBspOrganism() + " can't be parsed to integer", e);
+            }
+
+            logger.info("Returning info for kit w/ label " + kitLabel + " for " + bspKitQueryResult.getInstanceName());
+            return new BSPKitInfo(bspKitQueryResult.getBspCollection(),
+                    bspOrganism,
+                    "U",
+                    bspParticipantId,
+                    bspSampleId,
+                    bspMaterialType,
+                    bspReceptacleType);
+        }
+        else {
+            throw new RuntimeException("No participant id for " + kitLabel + " from " + bspKitQueryResult.getInstanceName());
+        }
+    }
+
+    private static void writeSampleReceivedToES(DDPInstance ddpInstance, BSPKitQueryResult bspKitInfo) {
+        String kitRequestId = new KitRequestDao().getKitRequestIdByBSPParticipantId(bspKitInfo.getBspParticipantId());
+        Map<String, Object> nameValuesMap = new HashMap<>();
+        nameValuesMap.put(ESObjectConstants.RECEIVED, SystemUtil.getISO8601DateString());
+        ElasticSearchUtil.writeSample(ddpInstance, kitRequestId, bspKitInfo.getDdpParticipantId(), ESObjectConstants.SAMPLES,
+                ESObjectConstants.KIT_REQUEST_ID, nameValuesMap);
+    }
+
+    private static void triggerDDP(Connection conn, @NonNull BSPKitQueryResult bspKitInfo, boolean firstTimeReceived, String kitLabel) {
```

Comment:
the class is everything except `Dao`. https://en.wikipedia.org/wiki/Data_access_object